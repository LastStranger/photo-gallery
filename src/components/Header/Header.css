@import '../../common_styles/colors.css';

.header-bar {
  background-color: gray(95%);
  color: gray(40%);
  border-bottom: 2px solid gray(85%);
  height: 12vh;
  display: flex;
  justify-content: space-around;
  align-items: center;
  font-weight: 500;
  position: fixed;
  top: 0;
  width: 100%;
  padding: 0 14%;

  &.header-bar-inverse {
    background-color: var(--nav-background);
    color: var(--icon-color);
    border-color: transparent;
  }
}

.header-title {
  font-size: 25px;
}

.header-container {
  transition: opacity .5s ease;
  opacity: 1;

  &.spacer {
    height: 12vh;
  }

  &.is-invisibile {
    transition: opacity .5s ease;
    opacity: 0;
  }
}

.back-icon {
  height: 18px;
  position: absolute;
  top: 5vh;
  right: 5%;
}
