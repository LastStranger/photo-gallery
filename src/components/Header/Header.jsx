// @flow

import { h } from 'preact';
import NavigationLink from '../NavigationLink';
import './Header.css';

const MAX_TITLE_LENGTH = 55;
const CHARS_CUT_OFF = 3;

const Header = (props: {
  title: string,
  showBackLink?: boolean,
  visibile?: boolean,
  invertColor?: boolean,
  showSpacer?: boolean,
  startLoading?: Function
}) =>
  <div className={`header-container${props.visibile ? '' : ' is-invisibile'}${props.showSpacer ? ' spacer' : ''}`}>
    <header className={`header-bar${props.invertColor ? ' header-bar-inverse' : ''}`}>
      <h1 className="header-title">
        {props.title.length <= MAX_TITLE_LENGTH
          ? props.title
          : `${props.title.substr(0, MAX_TITLE_LENGTH - CHARS_CUT_OFF)}â€¦`}
      </h1>
    </header>
    {props.showBackLink && props.visibile
      ? <NavigationLink onStart={props.startLoading} to="/">
          <img className="back-icon" alt="go back" src="/assets/icons/x.jpg" />
        </NavigationLink>
      : null}
  </div>;

Header.defaultProps = {
  showSpacer: false,
  showBackLink: false,
  visibile: true,
  invertColor: false,
  startLoading: () => {}
};

export default Header;
