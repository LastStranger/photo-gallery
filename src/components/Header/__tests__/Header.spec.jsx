// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import Header from '../';

test('renders default', () => {
  const string = render(<Header title="Image gallery" />);
  expect(string).toMatchSnapshot();
});

test('renders with inverted color', () => {
  const string = render(<Header invertColor title="Image gallery" />);
  expect(string).toMatchSnapshot();
});

test('renders with spacer', () => {
  const string = render(<Header showSpacer title="Image gallery" />);
  expect(string).toMatchSnapshot();
});

test('renders with back link', () => {
  const string = render(<Header showBackLink title="Image gallery" />);
  expect(string).toMatchSnapshot();
});

test('renders with invisible class', () => {
  const string = render(<Header visibile={false} title="Image gallery" />);
  expect(string).toMatchSnapshot();
});

test('renders with all props', () => {
  const string = render(<Header showSpacer showBackLink invertColor visibile={false} title="Image gallery" />);
  expect(string).toMatchSnapshot();
});
