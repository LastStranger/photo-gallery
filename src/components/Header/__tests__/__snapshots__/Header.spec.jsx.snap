// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<div class="header-container">
  <header class="header-bar">
    <h1 class="header-title">Image gallery</h1>
  </header>
</div>
`;

exports[`renders with all props 1`] = `
<div class="header-container is-invisibile spacer">
  <header class="header-bar header-bar-inverse">
    <h1 class="header-title">Image gallery</h1>
  </header>
</div>
`;

exports[`renders with back link 1`] = `
<div class="header-container">
  <header class="header-bar">
    <h1 class="header-title">Image gallery</h1>
  </header>
  <a class href="/">
    <img class="back-icon" alt="go back" src="/assets/icons/x.jpg" />
  </a>
</div>
`;

exports[`renders with inverted color 1`] = `
<div class="header-container">
  <header class="header-bar header-bar-inverse">
    <h1 class="header-title">Image gallery</h1>
  </header>
</div>
`;

exports[`renders with invisible class 1`] = `
<div class="header-container is-invisibile">
  <header class="header-bar">
    <h1 class="header-title">Image gallery</h1>
  </header>
</div>
`;

exports[`renders with spacer 1`] = `
<div class="header-container spacer">
  <header class="header-bar">
    <h1 class="header-title">Image gallery</h1>
  </header>
</div>
`;
